content/articles/how-to-write-efficient-python-code/index.md

```diff
@@ -0,0 +1,242 @@
+
+Let's go over the syntax of list comprehension;
+
+\**result*\* = [\**transform*\* \**iteration*\* \**filter*\*]
+
+The *result* would be the final list containing our data, the *transform* is the value of each data in our list, it keeps on changing as the value of the iterator changes. The *iteration* is a loop that helps populate our list with the required amount of data. Finally, the *filter* (optional) is used to filter out the data that we don't want.
+
+Before we take a look at an example of list comprehension, let's take a look at the traditional way of generating a list with data so we can compare the two ways.
+
+```python
+nums = []
+for x in range(10):
+    nums.append(x)
+
+print(nums)
+```
+
+**Output:**
+
+```bash
+[0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
+```
+It took us about 3 lines of code. Let's try to be more efficient using `list comprehension`.
+
+```python
+nums = [x for x in range(10)]
+
+print(nums)
+```
+
+**Output:**
+
+```bash
+[0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
+```
+
+Voilà, just one line of code to achieve the earlier result. It is way quicker to write once you understand the syntax and flow of `list comprehension`.
+
+Let's take a look at two more complex examples by adding *filters*.
+
+```python
+nums = [x for x in range(20) if x % 2 != 0]
+
+print(nums)
+```
+
+**Output:**
+
+```bash
+[0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
+```
+
+**Another Example**
+
+```python
+nums = ['True' if x % 2 == 0 else 'False' for x in range(10)]
+
+print(nums)
+```
+
+**Output:**
+
+```bash
+['True', 'False', 'True', 'False', 'True', 'False', 'True', 'False', 'True', 'False']
+```
+
+With these two examples above, we added the ability to use the `if/else` statement in our list comprehension. It should be noted that the `elif` statement can't be used within list comprehension.
+
+In this section, we learned about using list comprehension to generate our list for us with data with ease. In the next section, we will be taking a look at `lambda` functions.
+
+
+### Lambda Functions
+`Lambda` functions is a technique in Python of written our functions on a single line. They are regarded as anonymous (nameless) functions. They are also not suited for complex functions. Just like with list comprehension, they make our code small and concise.
+
+Let's take a look at the syntax;
+
+*lambda* arguments: expression
+
+The value at the left is referred to as the arguments, while the values at the right are the expression. Let's break the syntax further.
+
+*lambda* arguments: value_to_return *if* condition *else* value_to_return
+
+Before we take a look at an example of lambda functions, let's take a look at an example of a regular function in Python so we can compare the two ways.
+
+```python
+def even_or_odd(n):
+    if n % 2 == 0:
+        return "Even"
+    else:
+        return "Odd"
+
+print(even_or_odd(10))
+```
+
+**Output:**
+
+```bash
+Odd
```

Comment:
Also, the output here is wrong, it should be Even. @zolomohan

Reply:
Don't worry about it! I believe you are a good peer reviewer and anyone can make a mistake. @zolomohan